### PR TITLE
Added expert mode

### PR DESCRIPTION
**Pull request by melegiul**
### Short description

Added expert mode for displaying additional features.

### Proposed changes

Enabling expert mode includes the following features:
- Page mirroring
- Recurring events
- Do-Not-Translate-Tag in page editor
- Page-based permissions
- XLIFF import/export
- Page filtering

### Additional context

Expert mode setting is saved per user account, additional expert-mode-checkbox for (region-)user supplied.
Expert mode does not add an additional role to database, because expert mode does not deal with any additional permissions when enabled. Disabling expert mode just simplifies the GUI by hiding the aforementioned features.

### Resolved issues

Fixes: #499

### Screenshot

expert_mode

### CODE REVIEW

**Comment by timoludwig** [comment]
@melegiul Thanks for your PR!
I'm sorry, it turned out that our requirements for the expert mode were not well thought out and need some further discussions. Thus, I think we should conclude the discussion in Mattermost and issue #499 before proceeding this this PR. Sorry if this involves that you might have to discard parts of your work :confused:

**Comment by melegiul** [comment]
> @melegiul Thanks for your PR!
> I'm sorry, it turned out that our requirements for the expert mode were not well thought out and need some further discussions. Thus, I think we should conclude the discussion in Mattermost and issue #499 before proceeding this this PR. Sorry if this involves that you might have to discard parts of your work confused

No problem, I just tried to follow the discussion on mattermost, but i'm not sure if get it right, so it's probably good to list some open question.

**Comment by svenseeberg** [comment]
This is the result of the discussion (please correct me where I'm wrong):
* We consider translation experts a subgroup of expert users.
* We hide GUI elements relevant for translation from normal users, for example XLIFF import and export and filters.
* After we implemented all translation functionality (most important filtering), we iterate over the expert mode concept and could introduce an additional "translation expert role/mode". But this is not part of this issue/pull request.

**Comment by melegiul** [comment]
> This is the result of the discussion (please correct me where I'm wrong):
> 
>     * We consider translation experts a subgroup of expert users.
> 
>     * We hide GUI elements relevant for (mass-)translation from normal users, for example XLIFF import and export and filters.
> 
>     * After we implemented all translation functionality (most important filtering), we iterate over the expert mode concept and could introduce an additional "translation expert role/mode". But this is not part of this issue/pull request.

Yes, that should clarify the open questions, so I won't consider the topic about the "translation expert role/model" for now. That means an user operating in "expert mode" can not access XLIFF import/export elements, so I leave these elements for the next iteration, when the "translation expert mode" is implemented.

**Comment by timoludwig** [comment]
> That means an user operating in "expert mode" can not access XLIFF import/export elements, so I leave these elements for the next iteration, when the "translation expert mode" is implemented.

Just for reference: As we just discussed in our call, it is exactly the other way round: So users with "expert mode" should be able to access XLIFF functions and users without "expert mode" should not see this functions.
The next iteration is only optional if we determine that one mode is not enough and we need an additional setting.

File: src/cms/templates/users/admin/user.html [hunk 1]
```diff
@@ -83,6 +83,11 @@ <h3 class="heading font-bold text-white">{% trans 'Extended Settings' %}</h3>
                     {% render_field user_form.is_active %}
                 </div>
 
+                <div class="py-4 border-b solid border-gray-200">
+                    <label for="id_expert_mode" class="font-bold mr-2 cursor-pointer">{% trans 'Expert mode' %}</label>
+                    {% render_field user_profile_form.expert_mode %}
```
Comment (timoludwig) [inline src/cms/templates/users/admin/user.html:88]: I'd like to have an explanation here what features exactly are enabled with expert mode (and in the region user form as well).

**Review by timoludwig — COMMENTED** [review]
Thanks! In general, everything works as expected! :+1:
I just have one request regarding an explanation in the user form and a small improvement for the context.

File: src/cms/views/events/event_mixin.py [hunk 2]
```diff
@@ -23,3 +23,9 @@ class EventMixin(ContextMixin):
             "All translations of this event will also be deleted."
         ),
     }
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if "user" in kwargs:
+            context["expert_mode"] = kwargs["user"].profile.expert_mode
```
[the remark this answers is not in the document]
Comment (melegiul) [reply src/cms/views/events/event_mixin.py:30]: Well, then let's go with the `{% if request.user.profile.expert_mode %}` template, this seems to be more compact. I will remove the `get_context_data()` function entirely from both Mixins.

File: src/cms/templates/users/admin/user.html (hunk 1 above)
Comment (melegiul) [reply src/cms/templates/users/admin/user.html:88]: Alright, I added an explanation paragraph

**Review by timoludwig — COMMENTED** [review]
Looks good :+1: 

Sorry, I missed this earlier on: Could you also add the field `"expert_mode": true` to the test root user? This way, the expert mode is enabled by default for any new dev environments...
https://github.com/Integreat/integreat-cms/blob/ca8fc99d5e24a444891292c0811b5b3e772b7d2f/src/cms/fixtures/test_data.json#L2-L19

**Comment by melegiul** [comment]
> Looks good +1
> 
> Sorry, I missed this earlier on: Could you also add the field `"expert_mode": true` to the test root user profile? This way, the expert mode is enabled by default for any new dev environments...

No problem, expert mode is now enabled in test data root

**Review by timoludwig — APPROVED** [review]
Amazing, thanks! :rocket:

**Pull request merged**